### ISSUE TITLE
test_complete_hier_clusters_approx and test_clusters_approx return NaN for pval and stderr if run on datasets with more than 343 cols

### ISSUE THREAD

**Issue by Jannetty** [lucylgao/clusterpval#8]
Thank you for your work and for making this terrific R package!

I noticed when trying to run test_clusters_approx on the Zheng dataset from your [clusterpval-experiments](https://github.com/lucylgao/clusterpval-experiments) repo that test_complete_hier_clusters_approx and test_clusters_approx return NaN for pval and stderr on datasets with many columns. I think the `gamma(q/2)` on [line 278](https://github.com/lucylgao/clusterpval/blob/2bda312ea4b448ef86efb5adc792b28f61189661/R/trunc_inf.R#L278) (in test_complete_hier_clusters_approx) and [line 426](https://github.com/lucylgao/clusterpval/blob/2bda312ea4b448ef86efb5adc792b28f61189661/R/trunc_inf.R#L426) (in test_clusters_approx) will cause a float overflow if q is greater than 343 (if the input data has 344 or more columns). This float overflow causes the log_survives value to always be infinity, which I think makes the pval and stderr return NaN.

Because the gamma(q/2) value is immediately logged, and we know that q/2 will always be either a positive integer or a positive half integer (and thus we can calculate the particular values of the gamma function at each possible value of q/2), I think we can calculate the `log(gamma(q/2))` value ahead of time taking advantage of the fact that log(a*b) = log(a) + log(b) and log(a/b) = log(a) - log(b).

I wrote a few helper functions that calculate the `log(gamma(q/2))` values directly without overflowing a float (I will submit a pull request and link this issue). When using these helper functions both functions return float values for pval and stderr. Please let me know if I have misunderstood anything and thank you very much again for this R package!

**Comment by lucylgao** [comment]
Hi Sophia, 

Thank you for letting me know about this issue. I agree that overflow in the log(gamma(q/2)) call is very likely the issue! 

I do greatly appreciate your efforts in carefully crafting a solution to the problem yourself! However, I was able to find a stable and fast function for computing the log-gamma function in base R (lgamma). I've just pushed a new commit that uses lgamma(q/2) throughout instead of log(gamma(q/2)). Let me know if this doesn't fix your issue! 

Best, 
Lucy
